src/dataflow/operator.rs

```diff
@@ -43,10 +36,84 @@ where
     fn destroy(&mut self) {}
 }
 
-/*****************************************************************************
- * Sink: receives data with type T                                           *
- *****************************************************************************/
+/*************************************************************************************************
+ * ReadOnlySink: receives data with type T, and enables message callbacks to execute in parallel *
+ * by only allowing append access in the message callbacks and writable access to the state in   *
+ * the watermark callbacks.                                                                      *
+ ************************************************************************************************/
+
```

Comment:
Is this documentation? If so, we should comment with `///` so that the docs are properly generated.

Reply:
Nope, not documentation. Just a placeholder block to separate the different implementations. Will add documentation later once the PR is done.